### ISSUE TITLE
Translate template parameters

### ISSUE THREAD

**Issue by crep4ever** [patacrep/patagui#57]
How could we translate the templates options such as "title", "subtitle" etc. in the interface since they are generating from .tmpl files ?

**Comment by LaTruelle** [comment]
Je suis en train de modifier le code pour l'adapter aux changements dans patacrep, je suis tombé sur des choses qui me semblent étranges.

Est-ce que quelqu'un pourrait me dire à quoi sert le "qtpropertybrowser" qui est inclus? En particulier dans la classe CSongbook, au chargement/sauvegarde d'un songbook, il est utilisé mais je ne comprends pas trop à quoi sert-il...

cf. https://github.com/patacrep/patagui/blob/master/src/songbook.cc#L366-458 par exemple.

En mode debug, la partie dans le while n'est semble-t-il jamais atteinte, mais c'est peut-être que mes tests sont trop peu poussés ?

**Comment by Luthaf** [comment]
Alors là, il faut demander à ceux qui ont écrit ce code ! Ping @crep4ever @lohrun

**Comment by lohrun** [comment]
Salut,

Le qtpropertybrowser servait à afficher dynamiquement l'inferface de
configuration d'un songbook dépendant de ce que la partie JSON du template
décrit comme propriété. C'est donc que qui permet d'avoir une interface
dynamique qui s'adapte aux paramètres qui sont spécifiés dans le template.
Si je rajoute un paramètre dans le template, il est automatiquement
disponible dans l'interface.

J'espère que cela te donne une idée de l'utilité de cette classe.

Alexandre

**Comment by LaTruelle** [comment]
Salut,

OK je vois mieux l'utilité. Par contre, c'est effectivement utilisé ? Ou est-ce que c'est lié au format des templates qui a changé et du coup je ne vois pas la différence ?

Merci,
Emmanuel

**Comment by LaTruelle** [comment]
Salut,

Un petit up dans cette discussion... @crep4ever @lohrun, vous auriez un exemple de cas qui me permet de voir l'influence du qtpropertybrowser ?

Merci,
Emmanuel

**Comment by crep4ever** [comment]
Hello,

Le qtpropertybrowser servait pour l'onglet "songbook" du menu préférences.
Je l'avais désactivé depuis patacrep 4.0 car trop cassé.

en gros, ça parsait le fichier patacrep.tmpl (ou un autre fichier du genre)
et ça crée des widgets pour les valeurs style :
combobox instruments (guitare / ukulélé)
spinbox pour la taille de la police
checkbox pour les options images / accords / diagrammes d'accord
etc etc

C'est ce qui permettait de personnaliser le rendu du pdf en dehors de la
sélection des chansons.

**Comment by paternal** [comment]
> Je l'avais désactivé depuis patacrep 4.0 car trop cassé. en gros, ça parsait le fichier patacrep.tmpl (ou un autre fichier du genre)

Effectivement, on a changé des choses là dedans. Par contre, on a gardé l'idée de décrire les variables à l'intérieur des fichiers de templates, donc, en ré-écrivant la fonction qui va chercher les info, ça doit toujours être possible de faire ça.

**Comment by crep4ever** [comment]
Oui, ça "marchotait" encore mais un peu. Pour le réactiver, faut
décommenter les lignes:

diff --git a/src/preferences.cc b/src/preferences.cc
index d99de93..490354d 100644
--- a/src/preferences.cc
+++ b/src/preferences.cc
@@ -61,7 +61,7 @@ ConfigDialog::ConfigDialog(QWidget* parent)

   m_pagesWidget = new QStackedWidget(this);
   m_pagesWidget->addWidget(new OptionsPage(this));
-  //m_pagesWidget->addWidget(new SongbookPage(this));
+  m_pagesWidget->addWidget(new SongbookPage(this));
   m_pagesWidget->addWidget(new DisplayPage(this));
   m_pagesWidget->addWidget(new EditorPage(this));
 #ifdef ENABLE_LIBRARY_DOWNLOAD
@@ -103,11 +103,11 @@ void ConfigDialog::createIcons()
   optionsButton->setTextAlignment(Qt::AlignHCenter);
   optionsButton->setFlags(Qt::ItemIsSelectable | Qt::ItemIsEnabled);

-  //QListWidgetItem *songbookButton = new
QListWidgetItem(m_contentsWidget);
-  //songbookButton->setIcon(QIcon(":/icons/songbook/256x256/book.png"));
-  //songbookButton->setText(tr("Songbook"));
-  //songbookButton->setTextAlignment(Qt::AlignHCenter);
-  //songbookButton->setFlags(Qt::ItemIsSelectable | Qt::ItemIsEnabled);
+  QListWidgetItem *songbookButton = new QListWidgetItem(m_contentsWidget);
+  songbookButton->setIcon(QIcon(":/icons/songbook/256x256/book.png"));
+  songbookButton->setText(tr("Songbook"));
+  songbookButton->setTextAlignment(Qt::AlignHCenter);
+  songbookButton->setFlags(Qt::ItemIsSelectable | Qt::ItemIsEnabled);


et re-écrire la fonction CSongbook::changeTemplate() pour parser les
variables de patadata/templates/data.tex
